account_payment_purchase/models/purchase_order.py

```diff
@@ -56,7 +56,10 @@ def _compute_payment_mode(self):
 
     def _prepare_invoice(self):
         """Leave the bank account empty so that account_payment_partner set the
-        correct value with compute."""
+        correct value with compute and propagate payment mode to invoice."""
         invoice_vals = super()._prepare_invoice()
         invoice_vals.pop("partner_bank_id")
+        avoid_assign_pay_mode = self._context.get("avoid_pay_mode_propagate", False)
+        if not avoid_assign_pay_mode:
```

Comment:
Instead of having `not` and a "negative" variable, better to invert it (and the same for the rest):

```suggestion
        if assign_payment_mode:
```

Reply:
Reverted the variable, now it is possitive.